mcp-server/pkg/tools/alertsanalysis/alerts.go

```diff
@@ -0,0 +1,344 @@
+// Copyright 2025 Chronosphere Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package alertsanalysis provides tools for analyzing Chronosphere alerts.
+package alertsanalysis
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/go-openapi/strfmt"
+	"github.com/mark3labs/mcp-go/mcp"
+	"go.uber.org/zap"
+
+	"github.com/chronosphereio/chronosphere-mcp/generated/dataunstable/dataunstable"
+	"github.com/chronosphereio/chronosphere-mcp/generated/dataunstable/dataunstable/data_unstable"
+	"github.com/chronosphereio/chronosphere-mcp/generated/dataunstable/models"
+	"github.com/chronosphereio/chronosphere-mcp/mcp-server/pkg/tools"
+	"github.com/chronosphereio/chronosphere-mcp/mcp-server/pkg/tools/pkg/params"
+)
+
+var _ tools.MCPTools = (*Tools)(nil)
+
+type Tools struct {
+	logger          *zap.Logger
+	dataUnstableAPI *dataunstable.DataUnstableAPI
+}
+
+func NewTools(
+	dataUnstableAPI *dataunstable.DataUnstableAPI,
+	logger *zap.Logger,
+) (*Tools, error) {
+	logger.Info("alerts tool configured")
+
+	return &Tools{
+		logger:          logger,
+		dataUnstableAPI: dataUnstableAPI,
+	}, nil
+}
+
+func (t *Tools) GroupName() string {
+	return "alertsanalysis"
```

Comment:
meta: I want to make it clear that this is a bit different from the "alerts" source of truth (this is specifically for analytical access patterns), so I thought I'd be clear in that both in the tool name and package name